### ISSUE TITLE
Does this work with Hub firmware 206

### ISSUE THREAD

**Issue by bkonings** [materik/homebridge-harmonyhub#8]
It is unclear to me if this plugin is effected by the local connection change in firmware 206 of the harmony hub. I suspect it won't work?

**Comment by JotWee** [comment]
No, it works with 201 and stops working with 206

**Comment by bkonings** [comment]
This home assistant development seems to have figured out how to use the web sockets implementation.

https://github.com/iandday/pyharmony/pull/21/commits

**Comment by DJay-X** [comment]
@materik We appreciate your work. Is there a chance you find a way to get your plugin to work with the  trouble Logitech Harmony with >v.206 is giving us?
Thanks in advance for the time you are investing and your help.

**Comment by materik** [comment]
I haven't experienced any problems myself probably because I haven't upgrade to that Harmony version. All my interaction with Logitech Harmony is done through another npm package, so my guess is that he needs to release a fix.

**Comment by DJay-X** [comment]
> All my interaction with Logitech Harmony is done through another npm package

What npm package is this?
THX

**Comment by materik** [comment]
@DJay-X https://github.com/AirBorne04/harmonyhub/tree/master

**Comment by ak-enad** [comment]
Hi materik !

XMPP versus WebSocket ! :
https://www.jeedom.com/forum/viewtopic.php?f=142&t=41528&p=677221#p677221

akenad :)

**Comment by DJay-X** [comment]
Logitech is reacting and gives developers the following instruction.
I hope somebody will help the homebridge community to keep a harmony plugin working.
Thanks

> We’ve heard your concerns. We understand that some customers are frustrated with the recent security fix we put in place, as it closed access to private local API controls. While security continues to be a priority for us, we are working to provide a solution for those who still want access despite the inherent security risks involved.
> 
> If you would like to participate in an XMPP beta program, which will allow access to local controls, see the below instructions. Over the coming weeks, we will qualify a regular firmware release that still allows XMPP control for those who need it. We expect to send out an update that will be available to all Harmony customers in January.
> 
> Here are the instructions to access the program by updating the firmware on your Hub.
> 
> Launch the MyHarmony software on your desktop computer.
> From the login page, press the following keys to access the tool:
> On Windows - Press Alt + F9
> On Macs - Press Fn + Option + F9 or Option + F9.
> Scroll down to the bottom where it says “FIRMWARE TO ENABLE XMPP. FOR DEVELOPERS ONLY.”
> Be sure to read through the short warning and disclaimer to understand the impact of installing this firmware.
> Click on Update Firmware.
> Plug in your Harmony Hub via USB and click on Install.
> 
> Thank you

**Comment by nicoduj** [comment]
Hi, I tried a first plugin using websockets  to "counter" their removal of xmpp : 
https://www.npmjs.com/package/homebridge-harmony
https://github.com/nicoduj/homebridge-harmony
Feel free to try, at your own risks ! 
(code is still dirty, just a first try / work in progess)

**Comment by perlbo** [comment]
does this version work now with firmware 206 ?

**Comment by iharosi** [comment]
Hey @materik!

Check out @AirBorne04's comment here: https://github.com/AirBorne04/harmonyhub/issues/20#issuecomment-451682169
It seems like the `@harmonyhub/client` can be replaced with `@harmonyhub/client-ws`.

**Comment by materik** [comment]
@iharosi cool. thanks. will have a look as soon as i have time.

**Comment by materik** [comment]
Hi guys. Again, sorry for the slow response on my part. I've now changed the client dependency to `client-ws` so hopefully this will solve this firmware issue. Please test it out by installing the `develop` branch: ```npm install -g materik/homebridge-harmonyhub#develop``` and then run ```homebridge -D```. Post the log if you have any more issues. Thanks for all your help @iharosi @perlbo @ak-enad @nicoduj @DJay-X @bkonings

**Comment by nicoh88** [comment]
@materik 

**Works for me.** Thanks.

Only an error message during installation with `npmjs`, but without effects:

```
root@rpi-smarthome ~ # npm install -g materik/homebridge-harmonyhub#develop

> websocket@1.0.28 install /usr/lib/node_modules/homebridge-harmonyhub/node_modules/websocket
> (node-gyp rebuild 2> builderror.log) || (exit 0)

sh: 1: cannot create builderror.log: Permission denied
+ homebridge-harmonyhub@0.5.0
```

**Comment by materik** [comment]
@nicoh88 awesome! thanks a lot!

**Comment by AirBorne04** [comment]
hi @materik i just updated the package, due to an issue with the stateDigest processing. So give it a try with version 1.0.2 and it should be fine

**Comment by iharosi** [comment]
@materik @AirBorne04 great work guys! Thank you so much it works like a charm.

**Comment by materik** [comment]
@AirBorne04 thanks. will do. @iharosi thanks!

**Comment by zerogiant** [comment]
@materik switched to the develop build and everything appears to be working again with hub's 206 update

**Comment by materik** [comment]
Did update `client-ws` to 1.0.2 per @AirBorne04's comment and merged the PR to master so now `5.0.0` has the fixed version.
